### PR TITLE
Visual regression testing across document_types

### PR DESCRIPTION
**Pull request by theleebriggs**
# Motivation
__As__ a developer
__I want__ a up-to-date config of all document types in government-frontend
__So__ that I can use this for wraith testing with more confidence

Trello: https://trello.com/c/WqIzOp6X/358-1-make-wraith-rake-task-cover-all-document-in-government-frontend

# Description
This PR adds two new `rake` tasks:

`wraith:update_document_types` - generates a file of example pages and document_types rendered by `government_frontend`

`wraith:all_document_types` - runs visual regression tests across all examples found via `wraith:update_document_types`

# Example generated wraith config

http://axiomatic-reaction.surge.sh/wip-config-all-document-types.yaml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,15 +122,36 @@ bundle exec wraith latest test/wraith/config-examples.yaml
...
+This will run against the 10 (can be overidden with `:sample_size`) most popular pages as defined by the search api,
+for each known `document_type` in the app (see: [Generate a config for known document_types and example pages](#generate-a-config-for-known-document_types-and-example-pages)).
 
-If you wish to have your own local wip configs, wip* is in the .gitignore, so as an example
+*Note:* If you wish to have your own local wip configs, wip* is in the .gitignore, so as an example
 `wip-kittens.yaml` will be ignored
 
+#### Generate a config for known document_types and example pages
+
+Running the rake task below will retieve the `document_types` where `rendering_app = government_frontend` from the search api. It will then generate `test/wraith/wip-config-all-document-types.yaml`, this is a wraith config file containing the top 10 (can be overidden with `:sample_size`) example pages for each type.
```
Comment (fofr) [inline README.md]: Typo: retieve

File: lib/helpers/wraith_config_helper.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    config_paths = {}
+
+    if @paths.is_a? Hash
+      @paths.keys.each do |key|
+        @paths.fetch(key).each_with_index do |path, index|
+          config_paths["#{key}_#{index + 1}"] = path
+        end
+      end
+    else
+      @paths.each_with_index { |path, index| config_paths["#{name}_#{index + 1}"] = path }
```
Comment (fofr) [inline lib/helpers/wraith_config_helper.rb]: Minor: `config_paths["#{name}_#{index + 1}"] = path` is repeated, could be DRY'd up

File: lib/helpers/wraith_config_helper.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    end
+
+    config_paths
+  end
+
+  def write_config(config)
+    file_name = OUTPUT_PATH % { suffix: @name }
+    File.open(file_name, 'w') { |f| f.write config.to_yaml }
+    file_name
```
Comment (fofr) [inline lib/helpers/wraith_config_helper.rb]: Does `file_name` need to be returned?

**Comment by fofr** [comment]
Could you upload an example output to http://surge.sh/?

File: README.md (hunk 1 above)
Comment (fofr) [inline README.md]: `government_frontend` should be `government-frontend`

**Comment by fofr** [comment]
Couple of super minor things, otherwise :+1: :100:

File: README.md (hunk 1 above)
Comment (theleebriggs) [inline README.md]: Corrected.

File: lib/helpers/wraith_config_helper.rb (hunk 2 above)
Comment (theleebriggs) [inline lib/helpers/wraith_config_helper.rb]: DRY'd with a method.

File: lib/helpers/wraith_config_helper.rb (hunk 3 above)
Comment (theleebriggs) [inline lib/helpers/wraith_config_helper.rb]: Yes, as `File.open` will return the block result, which in this case is number of bytes written. I'd just have to move the `file_name` inside the open block, making for no code savings. At least that's how  I think it works.

**Comment by theleebriggs** [comment]
Updated description with `surge` link.

**Pull request merged**